terraform/modules/app/main.tf

```diff
@@ -62,16 +60,17 @@ resource "aws_ecs_service" "service" {
 }
 
 module "service_mesh_node" {
-  count = var.service_mesh ? length(local.container_services) : 0
+  # TODO: It looks like service_mesh (bool) is just a snowflake for grafana.
+  count = var.service_mesh ? 1 : 0
 
   source                           = "../service-mesh-node"
   backend_virtual_service_names    = var.backend_virtual_service_names
   mesh_name                        = var.mesh_name
-  port                             = local.container_services[count.index].port
-  protocol                         = local.container_services[count.index].protocol
+  port                             = var.port
+  protocol                         = "http"
   service_discovery_namespace_id   = var.service_discovery_namespace_id
   service_discovery_namespace_name = var.service_discovery_namespace_name
-  service_name                     = local.container_services[count.index].container_service
+  service_name                     = local.subdomain
```

Comment:
I know it is effectively the same, but I would have chosen, `var.service_name` rather than `local.subdomain`

Reply:
Good spot, I've changed this to `var.service_name`.